src/Okta.Sdk/CastingListAdapter{T}.cs

```diff
@@ -0,0 +1,88 @@
+﻿// <copyright file="CastingListAdapter{T}.cs" company="Okta, Inc">
+// Copyright (c) 2014-2017 Okta, Inc. All rights reserved.
+// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
+// </copyright>
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+
+namespace Okta.Sdk
+{
+    public sealed class CastingListAdapter<T> : IList<T>
+    {
+        private readonly IList<object> _genericList;
+        private readonly ILogger _logger;
+
+        public CastingListAdapter(IList<object> genericList, ILogger logger)
+        {
+            _genericList = genericList;
+            _logger = logger;
```

Comment:
I'm following the [style](https://github.com/aspnet/Home/wiki/Engineering-guidelines#coding-guidelines) that Microsoft uses for ASP.NET Core: `_camelCase` for private fields.